ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -203,58 +203,117 @@ public static void justificationAndFinalization(BeaconState state) throws Except
       // Case 2: epochs_since_finality > 4:
     } else if (epochs_since_finality.compareTo(FOUR) > 0) {
 
-      Function<Integer, UnsignedLong> inactivity_penalty =
-          new Function<Integer, UnsignedLong>() {
-            @Override
-            public UnsignedLong apply(Integer index) {
-              return inactivity_penality(
-                  state, index, epochs_since_finality, previous_total_balance);
-            }
+      Predicate<Integer> active_validators_filter =
+          index -> {
+            return ValidatorsUtil.is_active_validator_index(
+                state, index, BeaconStateUtil.get_current_epoch(state));
           };
 
-      Function<Integer, UnsignedLong> base_penalty =
-          new Function<Integer, UnsignedLong>() {
-            @Override
-            public UnsignedLong apply(Integer index) {
-              return base_reward(state, index, previous_total_balance);
-            }
+      Predicate<Integer> slashed_filter =
+          index -> {
+            Validator validator = state.getValidator_registry().get(index);
+            return validator.getSlashed_epoch().compareTo(BeaconStateUtil.get_current_epoch(state))
+                <= 0;
           };
 
       // prev epoch justified attester
       List<Integer> validator_indices =
```

Comment:
Is the below logic stubbed for something else? It looks like validator_indices is just being set to different references without anything actually being performed.

Reply:
It's a lambda function that collection.filter uses